lib/pure/strutils.nim

```diff
@@ -1456,28 +1456,82 @@ proc formatFloat*(f: float, format: FloatFormatMode = ffDefault,
   ## after the decimal point for Nim's ``float`` type.
   result = formatBiggestFloat(f, format, precision, decimalSep)
 
-proc formatSize*(bytes: BiggestInt, decimalSep = '.'): string =
-  ## Rounds and formats `bytes`. Examples:
+proc trimZeros*(x: string): string {.noSideEffect.} =
```

Comment:
proc trimZeros*(x: var string) {.noSideEffect.}

is more efficient and what you're doing is already slow enough. ;-)